### ISSUE TITLE
Guard stuck at Command execution: emacsclient  --eval '1' 2> /dev/null || echo 'N/A'

### ISSUE THREAD

**Issue by toctan** [guard/guard#477]
When the Emacs server is not running, Guard will get stuck:

``` bash
$ guard -d     
10:14:10 - DEBUG - Command execution: which notify-send
10:14:10 - DEBUG - Command execution: emacsclient --eval '1' 2> /dev/null || echo 'N/A'
```

I figured out with the environment variable `ALTERNATE_EDITOR` set to `vim`, the command

``` bash
emacsclient --eval '1' 2> /dev/null || echo 'N/A'
```

will end up opening vim! After hitting `:q!`, guard will start normally.

**Comment by vitaliel** [comment]
Maybe it can be better to not run ```Notifier.connect(state.session.notify_options)``` if notify is disabled.

**Comment by e2** [comment]
@vitaliel , @toctan - I don't use emacs, but I understand how frustrating it is when something doesn't work and there's no error.

I think I've fixed this with this patch in Notiffany: https://github.com/guard/notiffany/pull/15

Can you review/confirm this fix?

You can use this branch like so:

```ruby
gem 'notiffany', branch: 'e2-avoid_hanging_when_no_emacs_server'
```

Please let me know if this works, so I can release a fixed Notiffany to help other Emacs users out there - or give me a "recipe" on how to reproduce any problem you're having (I'm using Linux so this should be no problem).

@vitaliel - the Notifier has to always be connected and disconnected properly - even if notification is completely disabled, because notification can be disabled/enabled at runtime, some notifiers rely on cleaning up system resources (like TMux notifier needs to save the screen settings, change them, and then revert them before exiting), and it's also better to force the need to connect/disconnect properly, or you can end up with very, very hard to find bugs (especially since notifier settings are passed to child processes through the environment, like Guard::Cucumber's output formatter needs (needed?) to work from within the Cucumber process).

**Comment by vitaliel** [comment]
@e2 I had ALTERNATE_EDITOR env var, but after removing it, it works.
Your changes to override this env var is great.

**Comment by e2** [comment]
Fix released in notiffany 0.0.7
